Bottom: 1c160ef5df5740a60dc56e2b865de959ae19f00c
Top:    682d725f0f3e78b58304d69dab7b65bb75b6be16
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-06-27 09:56:41 -0700

Refresh of shut_wr

---

diff --git a/src/rsocket.c b/src/rsocket.c
index 581ecd8..4ecb425 100644
--- a/src/rsocket.c
+++ b/src/rsocket.c
@@ -1741,7 +1741,7 @@ int rshutdown(int socket, int how)
 	if (!rs_all_sends_done(rs) && !(rs->state & rs_error))
 		rs_process_cq(rs, 0, rs_all_sends_done);
 
-	if ((rs->fd_flags & O_NONBLOCK) && (how == SHUT_WR))
+	if ((rs->fd_flags & O_NONBLOCK) && (rs->state & rs_connected))
 		rs_set_nonblocking(rs, 1);
 
 	return 0;
